### PR TITLE
[chip, testplan] Update SYSRST_CTL and ADC

### PR DESCRIPTION
**Pull request by sriyerg**
- This commit addresses the comments and updates from the testplan
review meeting held on 6/07/2021.
- The meeting notes are below:
https://docs.google.com/document/d/1OhPP-HjciwKpIh0wWt1xqPqPf0Y0powmmww6xekwMeE/
- Updated SYSRST ctrl and ADC ctrl sections of the testplan

Signed-off-by: Srikrishna Iyer <sriyer@google.com>

### CODE REVIEW

**Review by msfschaffner — APPROVED** [review]
LGTM, but probably good if @esshiu can give this a pass as well.

**Review by udinator — APPROVED** [review]
LGTM, but will defer to @esshiu

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 1]
```diff
@@ -900,6 +901,173 @@
...
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sysrst_ctrl_outputs
+      desc: '''Verify that the sysrst_ctrl output pin values can be set.
+
+            - Drive a known value on ac_reset, ec_rst_l, pwrb and key* pins at the chip IO inputs.
+            - Verify that sysrst ctrl correctly loops them back to the chip IO outputs.
+            - Write the pin allowed ctl register to allow some of the pins to be overridden with
```
Comment (matutem) [inline hw/top_earlgrey/data/chip_testplan.hjson:921]: Consistently use ctrl
```suggestion
            - Write the pin allowed ctrl register to allow some of the pins to be overridden with
```

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 2]
```diff
@@ -900,6 +901,173 @@
...
+            - Verify that at the chip IO outputs, pins on which override should be active is
+              reflecting the overridden values. All others should reflect the values driven on chip
+              inputs.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sysrst_ctrl_in_irq
+      desc: '''Verify the SYSRST ctrl can detect an input combination to signal in interrupt.
```
Comment (matutem) [inline hw/top_earlgrey/data/chip_testplan.hjson:934]:
```suggestion
      desc: '''Verify the SYSRST ctrl can detect an input combination to signal an interrupt.
```

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 3]
```diff
@@ -900,6 +901,173 @@
...
+            - Verify separately, eack key combination sufccessfully generates an interrupt.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sysrst_ctrl_sleep_gsc_wakeup
+      desc: '''Verify the SYSRST ctrl can wake up the chip from deep sleep.
+
+            - Read the reset cause register in rstmgr to confirm we are in POR reset phase.
```
Comment (matutem) [inline hw/top_earlgrey/data/chip_testplan.hjson:952]: Why depend on coming from POR reset? This test could be run after some other test...

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 4]
```diff
@@ -900,6 +901,173 @@
...
+            - Program one of the com_sel_ctl_* CSRs to choose a set of inputs to be detected as
+              a low power wake up signal.
+            - Program the associated detection timer.
+            - Program the detection outcome CSR's (com_out_ctl) interrupt bit to 1.
+            - Program the pwrmgr to put the chip in deep sleep state and wake up on GSC wake up
```
Comment (matutem) [inline hw/top_earlgrey/data/chip_testplan.hjson:957]: Why call it GSC? is it not a sysrst_ctrl wake up? Is it historical?

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 5]
```diff
@@ -900,6 +901,173 @@
...
+              combination on inputs woke up the chip from sleep.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sysrst_ctrl_gsc_reset
+      desc: '''Verify the SYSRST ctrl can reset the chip from normal state.
+
+            - Read the reset cause register in rstmgr to confirm we are in POR reset phase.
```
Comment (matutem) [inline hw/top_earlgrey/data/chip_testplan.hjson:974]: Same as above... Imagine a version of this test run within a collection of other tests.

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 6]
```diff
@@ -900,6 +901,173 @@
...
+            - Program one of the com_sel_ctl_* CSRs to choose a set of inputs to be detected as
+              the GSC reset signal.
+            - Program the associated detection timer.
+            - Issue a WFI to to let the SW be in idle state.
```
Comment (matutem) [inline hw/top_earlgrey/data/chip_testplan.hjson:978]: Why should SW be idle for a reset?

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 7]
```diff
@@ -900,6 +901,173 @@
...
+      desc: '''Verify that in deep sleep, ADC ctrl can signal the ADC within the AST to power down.
+
+            - Read the reset cause register in rstmgr to confirm we are in POR reset phase.
+            - Follow the same steps as chip_adc_ctrl_debug_cable_irq, but instead of programming the
+              selected filter to interrupt, program it to wake up the chip from sleep.
+            - Program the pwrmgr to put the chip in deep sleep state and wake up on debug cable
+              detection.
+            - Issue a WFI to bring the chip in low power state.
+            - After some time, force the ADC output of AST to be a value within the programmed
+              filter range. That should cause the pwrmgr to be wake up from sleep.
```
Comment (matutem) [inline hw/top_earlgrey/data/chip_testplan.hjson:1059]:
```suggestion
              filter range. That should cause the pwrmgr to wake up.
```

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 1 above)
Comment (sriyerg) [reply hw/top_earlgrey/data/chip_testplan.hjson:921]: Aah, I mimicked the design which uses `ctl` in the CSR name.

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 7 above)
Comment (sriyerg) [reply hw/top_earlgrey/data/chip_testplan.hjson]: Fixed.

**Comment by esshiu** [comment]
Look good to me.

-Eric

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 4 above)
Comment (esshiu) [reply hw/top_earlgrey/data/chip_testplan.hjson:1063]: It's indeed historical. I got the name (GSC) from Randall.

**Pull request merged**